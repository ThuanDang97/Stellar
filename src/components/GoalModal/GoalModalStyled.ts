import styled from 'styled-components'

import { ViewStyled } from '@components/styled-components'
// Themes
import Metrics from '@themes/Metrics'

// Utils
import { pxToRem } from '@utils/theme'

export const WrapperGoalListContent = styled(ViewStyled)`
  @media (max-width: ${pxToRem(Metrics.breakPoints.lg)}) {
    overflow-y: scroll;
    max-height: ${pxToRem(420)};
  }
  @media (max-width: ${pxToRem(Metrics.breakPoints.md)}) {
    max-height: ${pxToRem(280)};
  }
`
