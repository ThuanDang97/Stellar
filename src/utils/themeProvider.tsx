import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
