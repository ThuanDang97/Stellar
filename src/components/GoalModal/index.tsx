import React from 'react'
import { useTheme } from 'styled-components'

// Components
import Button from '@components/Button'
import ListGoal from '@components/ListGoal'
import { TextStyled, ViewStyled } from '@components/styled-components'
import Title from '@components/Title'
import Modal from '@components/Modal'

// Props type
import { GoalModalProps } from '@self-types/components/GoalModal.props'
import { WrapperGoalListContent } from './GoalModalStyled'

const GoalModal = ({
  isOpen,
  onClose,
  selectedItem,
  setSelectedItem,
  selectedGoals,
  onClick,
}: GoalModalProps) => {
  const theme = useTheme()
  const hasSelectedGoal = selectedGoals && selectedGoals.length > 0

  const isDisabled =
    (!hasSelectedGoal && selectedItem?.length === 0) ||
    (hasSelectedGoal && selectedGoals?.length === selectedItem?.length)

  return (
    <Modal isOpen={isOpen} onClose={onClose} maxWidth={750}>
      <Title
        fontFamily="AdobeCleanExtraBold"
        color={theme.colors.black}
        title="What are your Stellar goals?"
        fontSize={theme.typography.fontSize.xlg}
      />
      <TextStyled
        mTop={theme.metrics.dimensions.md}
        mBottom={theme.metrics.dimensions.lg}
      >
        Select all that apply.
      </TextStyled>
      <ViewStyled display="flex" justifyContent="center" flexDirection="column">
        <WrapperGoalListContent>
          <ListGoal
            selectedItem={selectedItem}
            setSelectedItem={setSelectedItem}
            selectedGoals={selectedGoals}
          />
        </WrapperGoalListContent>
        <Button
          {...(!isDisabled && { cursor: 'pointer' })}
          title="Continue"
          type="submit"
          disabled={isDisabled}
          btnWidth={320}
          btnHeight={45}
          letterSpacing={theme.typography.letterSpacing.xl}
          onClick={onClick}
          mTop={theme.metrics.dimensions.lg}
          mBottom={theme.metrics.dimensions.md}
        />
      </ViewStyled>
    </Modal>
  )
}

export default React.memo(GoalModal)
