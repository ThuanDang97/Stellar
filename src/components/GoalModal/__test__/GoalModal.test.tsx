import React from 'react'
import { screen, cleanup } from '@testing-library/react'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Components
import GoalModal from '../index'

describe('GoalModal', () => {
  const props = {
    onClose: jest.fn(),
    selectedItem: [],
    setSelectedItem: jest.fn(),
    onSubmit: jest.fn(),
    selectedGoals: [],
  }
  beforeEach(() => {
    renderWithTheme(<GoalModal isOpen {...props} />)
  })

  afterEach(() => {
    cleanup()
  })

  it('should render GoalModal', () => {
    const backdrop = screen.getByTestId('backdrop')

    expect(backdrop).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { asFragment } = renderWithTheme(<GoalModal isOpen {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should disable btn', () => {
    cleanup()
    renderWithTheme(<GoalModal isOpen {...props} />)

    const button = screen.getByTestId('button')

    expect(button).toHaveAttribute('disabled')
  })

  it('should enable btn', () => {
    const newProps = { ...props, selectedItem: ['g1'] }

    cleanup()
    renderWithTheme(<GoalModal isOpen {...newProps} />)

    const button = screen.getByTestId('button')

    expect(button).not.toHaveAttribute('disabled')
  })
})
