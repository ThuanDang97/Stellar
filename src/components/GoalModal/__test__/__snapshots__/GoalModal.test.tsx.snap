// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalModal matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 ModalStyled__BackDropStyled-sc-d5vki3-1 fTsvHV cnMCGd"
    data-testid="backdrop"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 ModalStyled-sc-d5vki3-0 hzMsLt hTNOqC hmXhmG"
    >
      <div
        class="ModalStyled__ImageWrapper-sc-d5vki3-2 eSAkbC"
        data-testid="close-icon"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="close-icon"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <h1
        class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 fDhYUR WstJK"
        color="#000"
        font-family="AdobeCleanExtraBold"
        font-size="36"
      >
        What are your Stellar goals?
      </h1>
      <span
        class="TextStyled-sc-9xia41-0 fpvcHn"
      >
        Select all that apply.
      </span>
      <div
        class="ViewStyled-sc-rqe5sa-0 kruvYh"
        display="flex"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 GoalModalStyled__WrapperGoalListContent-sc-119mq2b-0 fTsvHV iZljWA"
        >
          <div
            class="ViewStyled-sc-rqe5sa-0 etuUCN"
            display="flex"
            width="100%"
          >
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Improve My 
 Credit Score
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Build 
 Wealth
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Buy A Car
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Get A Job
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Buy 
 A Home
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Rent 
 A Home
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Get A 
 Credit Card
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Get Better 
 Insurance Rate
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Get Better 
 Interest Rates
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Pay May Bills 
 On Time
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Get A Loan
              </p>
            </div>
            <div
              class="ViewStyled-sc-rqe5sa-0 ViewStyled__BoxShadowStyled-sc-rqe5sa-2 eBRJBV eFQizc"
              cursor="pointer"
              data-testid="card-goal"
              display="flex"
              height="120"
              width="160"
            >
              <div
                class="ViewStyled-sc-rqe5sa-0 CardGoalStyled__WrapperImage-sc-s2xlor-0 hQeZKG ihZIBO"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                >
                  <img
                    alt="Card Goal"
                    data-nimg="fill"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <p
                class="TextStyled-sc-9xia41-0 eTpfgb"
                color="#203F59"
                display="-webkit-box"
                font-size="14"
                overflow="hidden"
              >
                Start 
 A Business
              </p>
            </div>
          </div>
        </div>
        <button
          class="ButtonStyled-sc-1ys9v52-0 cyPVUL"
          data-testid="button"
          disabled=""
          letter-spacing="0.69"
          type="submit"
        >
          Continue
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
